packages/slate-locales-plugin/index.js

```diff
@@ -0,0 +1,183 @@
+const fs = require('fs-extra');
+const path = require('path');
+const {ConcatSource, RawSource} = require('webpack-sources');
+const _ = require('lodash');
+
+module.exports = class localesPlugin {
+  constructor(options) {
+    this.options = options;
+  }
+  apply(compiler) {
+    compiler.hooks.compilation.tap('Slate Locales Plugin', this.addLocales);
+  }
+
+  async addLocales(compilation) {
+    const dirents = await fs.readdir(
+      path.resolve(compilation.options.context, 'sections'),
+      {withFileTypes: true},
+    );
+    // eslint-disable-next-line prefer-arrow-callback
+    dirents.forEach((dirent) => {
+      if (dirent.isDirectory()) {
+        traverseDirectory(
+          dirent,
+          path.resolve(compilation.options.context, 'sections'),
+          compilation,
+          dirents,
+        );
+      } else if (dirent.isFile() && path.extname(dirent.name) === '.liquid') {
+        getSchema(
+          dirent,
+          path.resolve(compilation.options.context, 'sections'),
+          compilation,
+          dirents,
+        );
+      }
+    });
+  }
+};
+
+async function getSchema(liquidDirent, directoryPath, compilation, dirents) {
```

Comment:
This method is doing a lot more than just `getSchema` -- can you work on scoping the contents into logical methods to improve overall readability? The nesting is really hard to read.